x-pack/plugins/cases/public/common/hooks.ts

```diff
@@ -5,11 +5,120 @@
  * 2.0.
  */
 
+import { useCallback, useEffect, useState } from 'react';
+
+import { useLocation, useHistory } from 'react-router-dom';
+
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { parse, stringify } from 'query-string';
 import { STACK_APP_ID } from '../../common/constants';
 import { useCasesContext } from '../components/cases_context/use_cases_context';
 
+import { DEFAULT_QUERY_PARAMS } from '../containers/use_get_cases';
+
+import type {
+  LocalStorageQueryParams,
+  QueryParams,
+  ParsedUrlQueryParams,
+  UrlQueryParams,
+} from '../../common/ui/types';
+
+const parseUrlQueryParams = (search: string): UrlQueryParams => {
```

Comment:
How about we export this function and test it directly? I know that's not necessarily the best practice for functions that are really internal but it'd be good to have coverage over the parsing logic directly.

Reply:
will do 👍